dev/com.ibm.ws.kernel.service/src/com/ibm/ws/common/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,382 @@
+     */
+    public static String getSecureAlternative(String algorithm) {
+        return secureAlternative.get(algorithm);
+    }
+
+    public static String getSignatureAlgorithm() {
+        if (fipsEnabled && (isOpenJCEPlusFIPSAvailable() || isIBMJCEPlusFIPSAvailable()))
+            return SIGNATURE_ALGORITHM_SHA256WITHRSA;
+        else
+            return SIGNATURE_ALGORITHM_SHA1WITHRSA;
+    }
+
+    public static String getEncryptionAlgorithm() {
+        if (fipsEnabled && (isOpenJCEPlusFIPSAvailable() || isIBMJCEPlusFIPSAvailable()))
+            return ENCRYPT_ALGORITHM_RSA;
+        else
+            return ENCRYPT_ALGORITHM_DESEDE;
+    }
+
+    public static String getCipher() {
+        String cipher = fipsEnabled ? AES_GCM_CIPHER : DES_ECB_CIPHER;
+        return cipher;
+    }
+
+    public static void logInsecureAlgorithm(String configProperty, String insecureAlgorithm) {
+        // TODO remove beta check
+        if (isRunningBetaMode()) {
+            Tr.warning(tc, "CRYPTO_INSECURE", configProperty, insecureAlgorithm);
+        }
+    }
+
+    public static void logInsecureAlgorithmReplaced(String configProperty, String insecureAlgorithm, String secureAlgorithm) {
+        // TODO remove beta check
+        if (isRunningBetaMode()) {
+            Tr.warning(tc, "CRYPTO_INSECURE_REPLACED", configProperty, insecureAlgorithm, secureAlgorithm);
+        }
+    }
+
+    public static void logInsecureProvider(String provider, String insecureAlgorithm) {
+        // TODO remove beta check
+        if (isRunningBetaMode()) {
+            Tr.warning(tc, "CRYPTO_INSECURE_PROVIDER", provider, insecureAlgorithm);
+        }
+    }
+
+    public static boolean isIBMJCEAvailable() {
+        if (ibmJCEProviderChecked) {
+            return ibmJCEAvailable;
+        } else {
+            ibmJCEAvailable = JavaInfo.isSystemClassAvailable(IBMJCE_PROVIDER);
+            ibmJCEProviderChecked = true;
+            return ibmJCEAvailable;
+        }
+    }
+
+    public static boolean isIBMJCEPlusFIPSAvailable() {
+        if (ibmJCEPlusFIPSProviderChecked) {
+            return ibmJCEPlusFIPSAvailable;
+        } else {
+            ibmJCEPlusFIPSAvailable = JavaInfo.isSystemClassAvailable(IBMJCE_PLUS_FIPS_PROVIDER);
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "ibmJCEPlusFIPSProvider: " + IBMJCE_PLUS_FIPS_PROVIDER);
+            }
+
+            ibmJCEPlusFIPSProviderChecked = true;
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "ibmJCEPlusFIPSAvailable: " + ibmJCEPlusFIPSAvailable);
+            }
+
+            if (isRunningBetaMode() && ibmJCEPlusFIPSAvailable) {
+                ibmJCEPlusFIPSAvailable = true;
+            } else {
+                if (fipsEnabled && !isSemeruFips()) {
+                    Tr.error(tc, "FIPS is enabled but the " + IBMJCE_PLUS_FIPS_PROVIDER + " provider is not available.");
```

Comment:
Tr.error requires a translated message.  Is this temporary?

Reply:
This is temporary for now.